### PR TITLE
Add command line parameter for silent installations to disable NVDA at the logon screen and enhance portable-path command line behavior

### LINKED ISSUE #8574: Add a command line parameter to disable start at logon screen for silent installations
### Steps to reproduce:
1. Install NVDA with the --install-silent flag
2. Restart the system.

### Actual behavior:
NVDA runs at the logon screen.

### Expected behavior:
NVDA shouldn't run at the logon screen when explicitly chosen to install silent.

### Additional details
This issue is relevant for system administrators who want to install NVDA silently on a multi user system. IN the current situation, it is not possible to specify the start at logon behaviour from the command line. I think it makes sense to make False the default for first time silent installations. Alternatively, we could consider introducing a new command line parameter.

cc @michaeldcurran, @feerrenrut, @derekriemer

### System configuration:
Not applicable. This issue is unrelated to the Windows version of use and I believe it has been there since the  --install-silent parameter was introduced.

### PR DESCRIPTION
**Pull request by LeonarddeR**
### Link to issue number:
Fixes #8574

### Summary of the issue:
1. In the current situation, it is not possible to specify the start at logon behaviour from the command line. This issue is relevant for system administrators who want to install NVDA silently on a multi user system.
2. When starting NVDA with the `--portable-path` command line parameter and creating a portable copy from the GUI, the provided portable path isn't shown in the gui as one might expect.

### Description of how this pull request fixes the issue:
1. Added a `--enable-start-on-logon` command line parameter. It should be specified as `--enable-start-on-logon=True` or `--enable-start-on-logon=False`, just `--enable-start-on-logon` will show an error message. Alternative supported values are `yes/no`, `on/off`, `1/0` (i.e.  this option behaves as `configobj boolean`). NVDA's behaviour is to default this to `True`
2. When the `--portable-path` parameter is provided at NVDA start, the provided path is now pre-filled in in the portable copy creation dialog.

### Testing performed:
Ran try build.

1. Just the launcher: in the installation dialog, use NVDA on logon was enabled.
2. launcher with `--enable-start-on-logon=True`: in the installation dialog, use NVDA on logon was enabled.
3. launcher with `--enable-start-on-logon=False`: in the installation dialog, use NVDA on logon was disabled.
4. launcher with `--install-silent` and `--enable-start-on-logon=False`. IN the copy of NVDA that ran after installation, start on logon was disabled in the general settings.
5. Launcher with `--portable-path`: the provided portable path showed up in the create portable copy dialog.

### Known issues with pull request:
None

### Change log entry:
* New features
    - Added the `--disable-start-on-logon` command line parameter to allow silent installations of NVDA that don't run at the logon screen by default. (#8574)

* Changes
    - When NVDA is started with the `--portable-path` command line parameter, the provided path is automatically filled in when trying to create a portable copy of NVDA using the NVDA menu. (#8623)

### CODE REVIEW

**Comment by feerrenrut** [comment]
Hate to question this further at this stage. But, I'm wondering if it might be more future proof to have this option as `--enable-start-on-logon [true|false]`.

This means that the command line option remains useful if we ever change the default. It also allows people to be specific about their expectations on the command line (whether that is to enable or disable), rather than relying on the implicit default behaviour.

**Comment by leonardder** [comment]
I'm afraid having an enable command line parameter is problematic, as I 
think argparse is able to store either True or False, not True, False or 
None. In my opinion, enable-start-on-logon only makes sense if it really 
enables start on logon in every case. This is certainly something to 
explore though.

**Comment by feerrenrut** [comment]
If we are able to be concrete about it and say the default is `enable-start-on-logon=True` then I think it would be my preference, but first we have to identify what variations in behaviour exist at the moment (which I think is what you were getting at with your last comment).

Otherwise, it looks like you can use a combination of `default` and `type` to ensure a boolean arg with a default of None:

```
>>> import argparse
>>> p = argparse.ArgumentParser()
>>> p.add_argument('--foo', default=None, type=bool)
_StoreAction(option_strings=['--foo'], dest='foo', nargs=None, const=None, default=None, type=<type 'bool'>, choices=None, help=None, metavar=None)
>>> p.parse_args(['--foo', 'True'])
Namespace(foo=True)
>>> p.parse_args(['--foo', 'False'])
Namespace(foo=True)
>>> p.parse_args([])
Namespace(foo=None)
>>> p.parse_args(['--foo', '"hello"'])
Namespace(foo=True)
>>> p.parse_args(['--foo', '0'])
Namespace(foo=True)
>>> p.parse_args(['--foo', '1'])
Namespace(foo=True)
>>> p.parse_args(['--foo', 0])
Namespace(foo=False)
```

Note that this appears to rely on casting the argument value to bool. But I don't think this should be a problem.

**Comment by leonardder** [comment]
[**@feerrenrut**](https://github.com/feerrenrut) commented on [21 aug. 2018 05:36 CEST](https://github.com/nvaccess/nvda/pull/8623#issuecomment-414539338 "2018-08-21T03:36:48Z - Replied by Github Reply Comments"):
> Otherwise, it looks like you can use a combination of `default` and `type` to ensure a boolean arg with a default of None:

This works when throwing False or 0 at parse-args, but when providing 0 or False at the command line, it looks like it will still use the bool value of the string "0" or "False".

>>> bool("False")
True

We can probably fix this with an easy function though:

```
def stringToBool(s):
    return s.lower() not in ("0", "false", "off")
```

Then, use stringToBool as the type.

**Comment by derekriemer** [comment]
If you make it optional, with a default of None what happens?

**Comment by leonardder** [comment]
[**@derekriemer**](https://github.com/derekriemer) commented on [26 aug. 2018 08:28 CEST](https://github.com/nvaccess/nvda/pull/8623#issuecomment-416017253 "2018-08-26T06:28:56Z - Replied by Github Reply Comments"):
> If you make it optional, with a default of None what happens?

I updated the description of the pr to reflect your question.

**Comment by leonardder** [comment]
@feerrenrut: I think I've addressed your concerns. A user can now provide enable-start-on-logon=True or --enable-start-on-logon=False to forcefully enable or disable starting at the logon screen when installing or updating. The default for --enable-start-on-logon is None, in which case the old behaviour applies (i.e. if updating, use the current value, else True)

File: user_docs/en/userGuide.t2t [hunk 1]
```diff
@@ -2755,6 +2755,7 @@ Following are the command line options for NVDA:
 | None | --no-sr-flag  | Don't change the global system screen reader flag |
 | None | --install | Installs NVDA (starting the newly installed copy) |
 | None | --install-silent | Silently installs NVDA (does not start the newly installed copy) |
+| None | --enable-start-on-logon=True|False | When installing, enable NVDA's start on the logon screen |
```
Comment (feerrenrut) [inline user_docs/en/userGuide.t2t]: I think we could clarify that the default behaviour for a fresh install is to start NVDA on the logon screen, probably best to put that in the start at windows logon section. Then we link to the `StartAtWindowsLogon` section here:
```suggestion
| None | --enable-start-on-logon=True|False | When installing, enable NVDA's [start on the logon screen #StartAtWindowsLogon] |
```

**Review by feerrenrut — CHANGES_REQUESTED** [review]
Sorry, I forgot all about this PR. I'm generally happy with this, but have a suggestion to improve the user guide.

**Comment by leonardder** [comment]
Done, thanks for this. I added an additional sentence in the start at logon section.

**Review by feerrenrut — APPROVED** [review]

**Pull request merged**